### DISCUSSION TITLE
v2 feature preview and ETA?

### DISCUSSION THREAD

**Discussion by corysimmons** [remix-run/remix#3499] [General]
Remix seems right up my alley, but I'd like to hold off on using it until I know what's going to change in v2 and does the Remix team have a ballpark ETA on when v2 might release?

**Comment by kiliman** [comment]
Hmm.. not sure why you are concerned about a version 2.

Remix is production ready right now.

The team behind Remix and React Router are well-known for minimizing breaking changes or providing a clear migration path. See React Router v5 => v6 compat.

Are there specific features you're looking for?

**Reply by corysimmons** [reply]
I'd just prefer not to learn a new tool if its API is going to significantly change in a month or two. I'd rather just study something else and in a couple months study the new API. Specifically this [comment](https://github.com/remix-run/remix/discussions/478#discussioncomment-2555551) makes me feel like waiting.

>As we move the remix data APIs to react router

Like what APIs are going to move? Will I need to do a lot of work to upgrade an existing app? etc.

I'm really interested in Remix, and will almost certainly use it as much as I can, but this kind of _"spend my time studying something else now since the API might drastically change in a month or two"_  idea is valid.

**Reply by sergiodxa** [reply]
React Router is getting a lot of new APIs from Remix like Form, loaders, actions, useFetcher, etc. In the v6.4 (still in beta).

Once shipped, Remix will update the internal React Router dependency and start using those new RR features, this change will be internal.

The team mentioned multiple times they don’t plan a v2 soon, and if they did once they will give a smooth migration path.

No one wants an Angular.JS again.

**Reply by corysimmons** [reply]
Awesome thanks for answering, Sergio!
